aten/src/ATen/templates/TensorBody.h

```diff
@@ -1157,4 +1165,103 @@ inline c10::MaybeOwned<Tensor> Tensor::expect_contiguous(MemoryFormat memory_for
     return c10::MaybeOwned<Tensor>::owned(__dispatch_contiguous(memory_format));
   }
 }
+
+struct ViewMeta {
+  // The names of all existing view operators.
+  enum class Type {
```

Comment:
Why do you need to do this?
This feels like it will make adding new views even harder?
Is there any reason why the strategy used by the autograd view tracking is not possible here?
- Use size/stride info + as_strided for strided Tensors and non-cross-dtype views
- For other things, use a lambda that captures all the necessary arguments for the current function.

The ViewMeta you have here will need to get specialized for every single view function we have to be able to capture its arguments no?

Reply:
The new PR uses lambdas instead of a big enum, I think the amount of boilerplate per view op is (relatively) minimal but lmk what you think!